### PR TITLE
Debug TT draws

### PR DESCRIPTION
**Pull request by AnastaZIuk**

### CODE REVIEW

**Comment by AnastaZIuk** [comment]
TODO:

debug line & AABB draws

File: include/nbl/scene/ITransformTree.h [hunk 1]
```diff
@@ -156,9 +238,72 @@ class ITransformTree : public virtual core::IReferenceCounted
...
+				localGPUMemoryReqs.vulkanReqs.size = m_debugLiveAllocations.size() * sizeof(node_t);
+				localGPUMemoryReqs.mappingCapability = video::IDriverMemoryAllocation::EMCAF_READ_AND_WRITE;
+				m_debugLiveAllocationsGpuBuffer = std::move(device->createGPUBufferOnDedMem(localGPUMemoryReqs, true));
+
+				std::vector<node_t> debugLiveAllocationsData;
+				std::for_each(m_debugLiveAllocations.begin(), m_debugLiveAllocations.end(), [&debugLiveAllocationsData](const node_t& node) { debugLiveAllocationsData.emplace_back(node); });
+				commandBuffer->updateBuffer(m_debugLiveAllocationsGpuBuffer.get(), 0, m_debugLiveAllocationsGpuBuffer->getSize(), debugLiveAllocationsData.data());
+			}
+
+			//if (needFlush) what about this one?
```
Comment (AnastaZIuk) [inline include/nbl/scene/ITransformTree.h:274]: @Devsh-Graphics-Programming/seniors how can I check for flush need?

Comment (AnastaZIuk) [reply include/nbl/scene/ITransformTree.h]: not needed

File: include/nbl/builtin/glsl/transform_tree/debug/debug_draw.frag [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#version 440 core
+
+layout(location = 0) in vec4 inColor;
+layout(location = 0) out vec4 outColor;
+
+void main()
+{
+	outColor = inColor;
+}
```
Comment (devshgraphicsprogramming) [inline include/nbl/builtin/glsl/transform_tree/debug/debug_draw.frag:9]: dont we have a shader like this somewhere already in `builtin/material` or `builtin/{Not GLSL}`

Comment (AnastaZIuk) [reply include/nbl/builtin/glsl/transform_tree/debug/debug_draw.frag:9]: probably yes, use it?

Comment (AnastaZIuk) [reply include/nbl/builtin/glsl/transform_tree/debug/debug_draw.frag:9]: I will replace it after merging this stuff

**Pull request merged**